### PR TITLE
Parameterize: full dihedral angle parallelization

### LINKED ISSUE #517: Parallelization across dihedrals

### PR DESCRIPTION
**Pull request by raimis**
- [x] Refactor QM classes
- [x] Simultaneous QM job submission

Depends on #521 
Fixes #517

### CODE REVIEW

**Comment by raimis** [comment]
@stefdoerr 

This is the next PR requiring reviewing and merging!

**Comment by raimis** [comment]
Oh, wait @stefdoerr, I will merge master first.

File: htmd/parameterization/ffmolecule.py [hunk 1]
```diff
@@ -293,28 +293,43 @@ def fitDihedrals(self, dihedrals, geomopt=True):
...
-        qm_results = []
-        for dihedral, molecule in zip(dihedrals, molecules):
-            name = "%s-%s-%s-%s" % tuple(self.name[dihedral])
-            fitdir = os.path.join(self.outdir, dirname, name, self.output_directory_name())
-            os.makedirs(fitdir, exist_ok=True)
+        # Create directories for QM data
+        directories = []
+        dihedral_directory = 'dihedral-opt' if geomopt else 'dihedral-single-point'
+        for dihedral in dihedrals:
+            dihedral_name = '%s-%s-%s-%s' % tuple(self.name[dihedral])
```
Comment (stefdoerr) [inline htmd/parameterization/ffmolecule.py]: Try to use .format() for formatting strings. More python3-style. The % formatting is considered old-style.

**Comment by stefdoerr** [comment]
Minus my one comment on string formatting it looks fine. You want me to merge or do you want to change the formatting first?

**Comment by raimis** [comment]
@stefdoerr I will change the formatting. Thanks!

**Pull request merged**